crates/primitives/src/receipt.rs

```diff
@@ -654,7 +471,7 @@ mod tests {
     #[cfg(feature = "optimism")]
     #[test]
     fn decode_deposit_receipt_regolith_roundtrip() {
-        let data = hex!("7ef9010c0182b741b9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c0833d3bbf");
+        let data = hex!("b901107ef9010c0182b741b9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c0833d3bbf");
```

Comment:
had to update those tests because they were attempting to us `ReceiptWithBloom::decode` on EIP-2718 encoding while it expects network encoding

this was working before because the decoding impl wasn't strict enough but this kind of encoding shouldn't really happen at all as receipts are only being decoded in p2p